hypervisor/src/kvm/mod.rs

```diff
@@ -3058,6 +3065,30 @@ impl KvmVcpu {
             .set_vcpu_events(events)
             .map_err(|e| cpu::HypervisorCpuError::SetVcpuEvents(e.into()))
     }
+
+    /// Get the state of the nested guest from the current vCPU,
+    /// if there is any.
+    #[cfg(target_arch = "x86_64")]
+    fn nested_state(&self) -> cpu::Result<Option<KvmNestedStateBuffer>> {
+        let mut buffer = KvmNestedStateBuffer::empty();
+
+        self.fd
+            .lock()
+            .unwrap()
+            .get_nested_state(&mut buffer)
+            .map(|size| size.map(|_| buffer))
```

Comment:
It took me too long to figure out what this line is doing. Please add an `Ok(buffer)` at the end and drop this. It's a bit too clever :-)